From 345793e6a4610f0fe28186a63f127a5a6880e549 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michel=20D=C3=A4nzer?= <mdaenzer@redhat.com>
Date: Tue, 8 Sep 2026 16:15:44 +0200
Subject: [PATCH] clutter/frame-clock: Preserve pending_reschedule from
 _schedule_update*

If clutter_frame_clock_schedule_update set pending_reschedule to TRUE
(most likely because the frame clock is inhibited), we would immediately
set it to FALSE again, which could cause malfunction.

Pointed out by Sebastian Keller.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/5310>
---
 clutter/clutter/clutter-frame-clock.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/clutter/clutter/clutter-frame-clock.c b/clutter/clutter/clutter-frame-clock.c
index f13f1f6202..d00c414712 100644
--- a/clutter/clutter/clutter-frame-clock.c
+++ b/clutter/clutter/clutter-frame-clock.c
@@ -477,12 +477,14 @@ maybe_reschedule_update (ClutterFrameClock *frame_clock)
     {
       if (frame_clock->pending_reschedule_now)
         {
+          frame_clock->pending_reschedule = FALSE;
           frame_clock->pending_reschedule_now = FALSE;
           clutter_frame_clock_schedule_update_now (frame_clock);
         }
       else if (frame_clock->pending_reschedule ||
                !frame_clock->throttle_timelines)
         {
+          frame_clock->pending_reschedule = FALSE;
           frame_clock->throttle_timelines = TRUE;
           clutter_frame_clock_schedule_update (frame_clock);
         }
@@ -491,7 +493,6 @@ maybe_reschedule_update (ClutterFrameClock *frame_clock)
           clutter_frame_clock_schedule_update_throttled (frame_clock);
         }
 
-      frame_clock->pending_reschedule = FALSE;
       return;
     }
 
-- 
2.55.0
